test/steps/kogitoexplainability.go

```diff
@@ -0,0 +1,128 @@
+}
+
+func (data *Data) installKogitoExplainabilityServiceWithReplicasWithConfiguration(replicas int, table *godog.Table) error {
+	explainability := &bddtypes.KogitoServiceHolder{
+		KogitoService: framework.GetKogitoExplainabilityResourceStub(data.Namespace, replicas),
+	}
+
+	if err := mappers.MapKogitoServiceTable(table, explainability); err != nil {
+		return err
+	}
+
+	return framework.InstallKogitoExplainabilityService(data.Namespace, framework.GetDefaultInstallerType(), explainability)
+}
+
+func (data *Data) kogitoExplainabilityHasPodsRunningWithinMinutes(podNb, timeoutInMin int) error {
+	return framework.WaitForKogitoExplainabilityService(data.Namespace, podNb, timeoutInMin)
+}
+
+type executionsResponse struct {
+	Executions []execution `json:"headers"`
+}
+
+type execution struct {
+	ExecutionID string `json:"executionId"`
+}
+
+func (data *Data) explainabilityResultIsAvailable(timeoutInMin int) error {
+	// Retrieve the execution id from the trusty service
+	executionsPath := "executions"
+	responseContent := "DECISION"
+	trustyServiceName := "trusty"
+	uri, err := framework.WaitAndRetrieveEndpointURI(data.Namespace, trustyServiceName)
+	if err != nil {
+		return err
+	}
+
+	executionsPath = data.ResolveWithScenarioContext(executionsPath)
+	requestInfo := framework.NewGETHTTPRequestInfo(uri, executionsPath)
+	responseContent = data.ResolveWithScenarioContext(responseContent)
+	err = framework.WaitForOnOpenshift(data.Namespace, fmt.Sprintf("GET request on path %s to return response content '%s'", executionsPath, responseContent), timeoutInMin,
+		func() (bool, error) {
+			return framework.DoesHTTPResponseContain(data.Namespace, requestInfo, responseContent)
+		})
+
+	if err != nil {
+		return err
+	}
+
+	response, err := framework.ExecuteHTTPRequest(data.Namespace, requestInfo)
+	if err != nil {
+		return err
+	}
+
+	executionsResponse := new(executionsResponse)
+	err = getJSON(response, &executionsResponse)
+	if err != nil {
+		return err
+	}
+
+	executionID := executionsResponse.Executions[0].ExecutionID
+
+	// Retrieve explainability result for the given execution ID
+	executionsPath = fmt.Sprintf("executions/decisions/%s/explanations/saliencies", executionID)
+	responseContent = "SUCCEEDED"
+
+	if err != nil {
+		return err
+	}
+	executionsPath = data.ResolveWithScenarioContext(executionsPath)
+	requestInfo = framework.NewGETHTTPRequestInfo(uri, executionsPath)
+	responseContent = data.ResolveWithScenarioContext(responseContent)
+	return framework.WaitForOnOpenshift(data.Namespace, fmt.Sprintf("GET request on path %s to return response content '%s'", executionsPath, responseContent), timeoutInMin,
+		func() (bool, error) {
+			return framework.DoesHTTPResponseContain(data.Namespace, requestInfo, responseContent)
+		})
+}
+
+func getJSON(r *http.Response, target interface{}) error {
```

Comment:
I think this method can be moved to framework/http.go. There are some places that could reuse this logic: keycloak.go (in master branch), and http.go too.